File: MATLAB/dir_excel_merge/excel_merge.m
function excel_merge()

    clear all;    

    addpath(genpath('lib'),'data');

    dataFile(1).name = select_data_file(1);
    if strcmp(dataFile(1).name, '!')
        return;
    end
    dataFile(2).name = dataFile(1).name;
    
    while strcmp(dataFile(1).name, dataFile(2).name)
        dataFile(2).name = select_data_file(2);
        if strcmp(dataFile(2).name, '!')
            return;
        end
        if strcmp(dataFile(1).name, dataFile(2).name)
            disp('Please select a second file that is different from the first. Hit any key to continue:');
            pause
        end        
    end % while strcmp()
    
    for i = 1:2
        dataFile(i).raw = read_data_file(dataFile(i).name, i);
    end
    
    disp(' ');
    disp('Merging data files ...');
    [dataFileMerged.raw, dataFile(1).HCID, dataFile(2).HCID, dataFileMerged.HCID, ...
        dataFile(1).exclusion, dataFile(2).exclusion, dataFileMerged.exclusion, ...
        dataFile(1).STID, dataFile(2).STID, dataFileMerged.STID] = merge_data_files(dataFile(1).raw, dataFile(2).raw);
  
    disp(' ');
    disp('Writing merged data to file ...');
    dataFileMerged.name = determine_merged_file_name(dataFile(1).name, dataFile(2).name);
    xlswrite(dataFileMerged.name, dataFileMerged.raw);

    add_highlights(dataFileMerged.name, dataFile(1), dataFile(2), dataFileMerged); 
    
    disp(' ');
    disp('Finis.');
    disp(' ');
    
end % excel_merge()

File: Python/dir_excel_merge/excel_merge.py
import pandas as pd

from lib.select_file import select_data_file
from lib.read_data import read_data_file
from lib.merge_data import merge_data_files
from lib.determine_name import determine_merged_file_name
from lib.write_data import write_data_file

dataFileName = [None] * 2

dataFileName[0] = select_data_file(1);
if dataFileName[0] == '!':
    exit()
#dataFileName[0] = './data/{}'.format(dataFileName[0])
dataFileName[1] = dataFileName[0]

while dataFileName[1] == dataFileName[0]:
    dataFileName[1] = select_data_file(2);
    if dataFileName[1] == '!':
        exit()
    if dataFileName[1] == dataFileName[0]:
        programPause = raw_input('Please select a second file that is different from the first. Hit the <ENTER> key to continue:')
#dataFileName[1] = './data/{}'.format(dataFileName[1])

dataFrames = {0: {}, 1: {}}
for i in range(2):
    dataFrames[i] = read_data_file(dataFileName[i], i)
                
print '\nMerging data files ...'
dataFrameMerged, dfHCID1, dfHCID2, dfHCIDboth, dfExclusion1, dfExclusion2, dfExclusionboth, dfSTID1, dfSTID2, dfSTIDboth = merge_data_files(dataFrames[0], dataFrames[1])

print '\nWriting merged data to file ...'
dataFileNameMerged = determine_merged_file_name(dataFileName[0], dataFileName[1])
write_data_file(dataFrameMerged, dfHCID1, dfHCID2, dfHCIDboth, dfExclusion1, dfExclusion2, dfExclusionboth, dfSTID1, dfSTID2, dfSTIDboth, dataFileNameMerged)

print '\nFinis.\n'